encrypted-data-model/src/main/java/ch/prevo/open/encrypted/services/EncryptedData.java

```diff
@@ -0,0 +1,131 @@
+package ch.prevo.open.encrypted.services;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.IOException;
+import java.io.Serializable;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.util.Base64;
+import java.util.Objects;
+
+/**
+ * Uses AES/RSA to encrypt data using a public key, and decryption using a private key.
+ *
+ * @param <T> type of the data to encrypt
+ */
+public abstract class EncryptedData<T> implements Serializable {
```

Comment:
So I'm still not a big fan of having this tight together 😸 an argument I have so far is, that it makes it harder to test. (random key generation is automatically applied)

Reply:
Would require us to have the private key available in the test; perhaps this should rather be tested in an integration test?

Reply:
So I changed this in spite of my earlier objections, as I found the mutability in that context to be less than useful.